frame/staking/src/pallet/impls.rs

```diff
@@ -161,10 +171,14 @@ impl<T: Config> Pallet<T> {
 				.with_weight(T::WeightInfo::payout_stakers_alive_staked(0))
 		})?;
 
-		let controller = Self::bonded(&validator_stash).ok_or_else(|| {
-			Error::<T>::NotStash.with_weight(T::WeightInfo::payout_stakers_alive_staked(0))
+		let account = StakingAccount::Stash(validator_stash.clone());
+		let mut ledger = Self::ledger(account.clone()).or_else(|_| {
+			if StakingLedger::<T>::is_bonded(account) {
+				Err(Error::<T>::NotController.into())
+			} else {
+				Err(Error::<T>::NotStash.with_weight(T::WeightInfo::payout_stakers_alive_staked(0)))
+			}
```

Comment:
I see the post dispatch weight is different for `NotController` and `NotStash` error but we could fix it to return same weights.

```suggestion
		let mut ledger = Self::ledger(account.clone())
			.or_else(|e| Err(e.with_weight(T::WeightInfo::payout_stakers_alive_staked(0))))?;
```